djangoProject3/DFS/views.py

```diff
@@ -0,0 +1,295 @@
+    queryset = Author.objects.all()
+    serializer_class = AuthorSerializer
+
+
+class GenreView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class BookView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    queryset = Book.objects.all()
+    serializer_class = BookSerializer
+    filter_backends = [SearchFilter]
+    search_fields = ['^title', '^genres__name', '^author__name']
+
+
+class ReviewView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    serializer_class = ReviewSerializer
+
+    def get_queryset(self):
+        book_id = self.request.query_params.get('book')
+        if book_id:
+            book_id = book_id.rstrip('/')
+            try:
+                book_id = int(book_id)
+            except ValueError:
+                return Review.objects.none()
+
+            return Review.objects.filter(book_id=book_id)
+
+        return Review.objects.all()
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class RatingView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    serializer_class = RatingSerializer
+
+    def get_queryset(self):
+        book_id = self.request.query_params.get('book')
+        if book_id:
+            book_id = book_id.rstrip('/')
+            try:
+                book_id = int(book_id)
+            except ValueError:
+                return Rating.objects.none()
+
+            return Rating.objects.filter(book_id=book_id)
+        return Rating.objects.all()
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class NotificationView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    queryset = Notification.objects.all()
+    serializer_class = NotificationSerializer
+
+
+class OrderView(viewsets.ModelViewSet):
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+
+
+class SearchView(APIView):
+    def get(self, request):
+        query = request.query_params.get('query', '')
+
+        search_results = Book.objects.filter(
```

Comment:
use ```django_filter``` package instead of manually doing everything

Reply:
by using django_filter it only search on the basis of title not for genres and author